### ISSUE TITLE
[StratCon] NPE - game freezes during firing phase

### ISSUE THREAD

**Issue by JonhyBurner** [MegaMek/mekhq#2733]
MekHQ 49.2

During a convoy protection scenario, both ally and enemy artillery forces deployed. I deployed a couple of my own.

After several turns with no issues, the game started freezing during the allied indirect fire phase. Kicking the bot and replacing didn't help. Exiting megamek and reloading, replacing the bot causes the same freeze. Reloading the save and using skip command works, until the next indirect fire phase. I tried waiting it out and eventually, the game moves forward, by apparently skipping the turn on its own?

logs (with NullPointerExceptions, I know you love those)
[megameklog.txt](https://github.com/MegaMek/mekhq/files/6746446/megameklog.txt)
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/6746448/mekhqlog.txt)

the saves, if you whish to try it out:
[Lisbon's Shadows Pre-Battle3.cpnx.gz](https://github.com/MegaMek/mekhq/files/6746454/Lisbon.s.Shadows.Pre-Battle3.cpnx.gz)
[customs.zip](https://github.com/MegaMek/mekhq/files/6746457/customs.zip)

[ingame4.sav.gz](https://github.com/MegaMek/mekhq/files/6746458/ingame4.sav.gz)
[ingame5.sav.gz](https://github.com/MegaMek/mekhq/files/6746459/ingame5.sav.gz)

the second save was made after the freeze

edit: apparently all firing phases freeze and right now the scenario is unplayable

**Comment by HammerGS** [comment]
FROM MegaMek Log
```
`ERROR` [megamek.server.Server] {AWT-EventQueue-0} 
loadGame(), line 1272 : Unable to load file: D:\Dados - Jogos\Game Installs\mekhq-windows-0.49.2\campaigns\Lisbon's Shadows Debriefing.cpnx.gz
com.thoughtworks.xstream.mapper.CannotResolveClassException: campaign
	at com.thoughtworks.xstream.mapper.DefaultMapper.realClass(DefaultMapper.java:81)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:125)
	at com.thoughtworks.xstream.mapper.DynamicProxyMapper.realClass(DynamicProxyMapper.java:55)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:125)
	at com.thoughtworks.xstream.mapper.PackageAliasingMapper.realClass(PackageAliasingMapper.java:88)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:125)
	at com.thoughtworks.xstream.mapper.ClassAliasingMapper.realClass(ClassAliasingMapper.java:79)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:125)
	at com.thoughtworks.xstream.mapper.ArrayMapper.realClass(ArrayMapper.java:74)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:125)
	at com.thoughtworks.xstream.mapper.SecurityMapper.realClass(SecurityMapper.java:71)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:125)
	at com.thoughtworks.xstream.mapper.CachingMapper.realClass(CachingMapper.java:47)
	at com.thoughtworks.xstream.core.util.HierarchicalStreams.readClassType(HierarchicalStreams.java:29)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.start(TreeUnmarshaller.java:133)
	at com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.unmarshal(AbstractTreeMarshallingStrategy.java:32)
	at com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1404)
	at com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1383)
	at com.thoughtworks.xstream.XStream.fromXML(XStream.java:1277)
	at megamek.server.Server.loadGame(Server.java:1270)
	at megamek.server.Server.loadGame(Server.java:1250)
	at mekhq.MekHQ.startHost(MekHQ.java:479)
	at mekhq.gui.BriefingTab.loadScenario(BriefingTab.java:570)
	at mekhq.gui.BriefingTab.lambda$initTab$10(BriefingTab.java:267)
```

FROM MekHQ Log
Lots of these
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at megamek.client.ui.swing.boardview.BoardView1.getArtilleryAttacksAtLocation(BoardView1.java:6139)
	at megamek.client.ui.swing.boardview.BoardView1.getHexTooltip(BoardView1.java:6028)
	at megamek.client.ui.swing.boardview.BoardView1$2.mouseMoved(BoardView1.java:630)
```

**Comment by JonhyBurner** [comment]
Just to add that somehow, sometime along the way, the hex tooltips stopped working as well

**Comment by NickAragua** [comment]
Fixed with megamek PR #3011
